rules/CodingStyle/ClassNameImport/ClassNameImportSkipVoter/ClassLikeNameClassNameImportSkipVoter.php

```diff
@@ -33,9 +35,31 @@ public function shouldSkip(File $file, FullyQualifiedObjectType $fullyQualifiedO
             return false;
         }
 
+        /**
+         * Note: Don't use ScopeFetcher::fetch() on Name instance,
+         * Scope can be null on Name
+         * This is part of ScopeAnalyzer::NON_REFRESHABLE_NODES
+         * @see https://github.com/rectorphp/rector-src/blob/9929af7c0179929b4fde6915cb7a06c3141dde6c/src/NodeAnalyzer/ScopeAnalyzer.php#L17
+         */
+        $scope = $node->getAttribute(AttributeKey::SCOPE);
+        $namespace = $scope instanceof Scope ? $scope->getNamespace() : null;
+        $namespace = strtolower((string) $namespace);
+
         $shortNameLowered = $fullyQualifiedObjectType->getShortNameLowered();
+        $fullyQualifiedObjectTypeNamespace = strtolower(
+            substr($fullyQualifiedObjectType->getClassName(), 0, -strlen($fullyQualifiedObjectType->getShortName()) - 1)
```

Comment:
@samsonasik is the last `-1` really correct?

Reply:
I will look into it